File: Java Labs/Patterns On Java/src/com/pattern/decorator/ColorDecorator.java
package com.pattern.decorator;

public class ColorDecorator extends ClothingItemDecorator {
    private String color;

    public ColorDecorator(ClothingItem decoratedItem, String color) {
        super(decoratedItem);
        this.color = color;
    }

    @Override
    public String getDescription() {
        return super.getDescription() + " Color: " + color;
    }

    @Override
    public double getPrice() {
        // Additional cost for color
        return super.getPrice() + 5.0;
    }
}

File: Python Labs/com/pattern/decorator/ColorDecorator.py
from com.pattern.decorator.ClothingItemDecorator import ClothingItemDecorator


class ColorDecorator(ClothingItemDecorator):
    def __init__(self, decorated_item, color):
        super().__init__(decorated_item)
        self.color = color

    def get_description(self):
        return super().get_description() + f" Color: {self.color}"

    def get_price(self):
        return super().get_price() + 5.0  # Additional cost for color